common/persistence/health_signal_aggregator.go

```diff
@@ -150,12 +157,21 @@ func (s *HealthSignalAggregatorImpl) emitMetricsLoop() {
 			return
 		case <-s.emitMetricsTimer.C:
 			s.requestsLock.Lock()
-			requestCounts := s.requestsPerShard
-			s.requestsPerShard = make(map[int32]int64, len(requestCounts))
+			requestCounts := s.requestCounts
+			s.requestCounts = make(map[int32]map[string]int64, len(requestCounts))
 			s.requestsLock.Unlock()
 
-			for shardID, count := range requestCounts {
-				shardRPS := int64(float64(count) / emitMetricsInterval.Seconds())
+			for shardID, requestCountPerNS := range requestCounts {
+				shardRequestCount := int64(0)
+				for namespace, count := range requestCountPerNS {
+					shardRequestCount += count
+					shardRPSPerNS := int64(float64(count) / emitMetricsInterval.Seconds())
+					if shardRPSPerNS > int64(s.perShardPerNsRPSWarnLimit()) {
```

Comment:
is it possible to default ShardPerNsRPSWarnLimit to a percentage of ShardRPSWarnLimit, like 80%? say if the dc value is <= 0 then fallback to 80% of ShardRPSWarnLimit.

Reply:
I changed it to a percentage with a default of 80% when not set. I made it so that if the value is <= 0 then per-shard-per-ns logs are not emitted. It might be more useful to disable them if they are noisy then just always default to 80%.